src/cmdstan/arguments/arg_hmc.hpp

```diff
@@ -17,10 +16,15 @@ class arg_hmc : public categorical_argument {
     _description = "Hamiltonian Monte Carlo";
 
     _subarguments.push_back(new arg_engine());
-    _subarguments.push_back(new arg_metric());
-    _subarguments.push_back(new arg_metric_file());
-    _subarguments.push_back(new arg_stepsize());
-    _subarguments.push_back(new arg_stepsize_jitter());
+    _subarguments.push_back(
+        new arg_single_string("metric", "Geometry of base manifold", "diag_e"));
```

Comment:
there's a check in the CmdStan code later to that effect, no?

Reply:
I can't seem to find one. I believe if `metrix->value()` ends up returning a string that isn't one of those 3 choices, no sampling is done _and_ no error is raised.

Reply:
lines #503-#509.

Reply:
Ah, I was still looking in the `command.hpp` before this PR. My bad!

I think I would still prefer to preserve the existing logic so the check happens at argument parse time, rather than after the model has already been instantiated.

Reply:
OK.
we need class `arg_single_string_choices` will need to do two things:
1.  add vector of allowed values
2. override the `parse_args` method.

Reply:
Are there other things we would use it for? If not I’d be just as happy to leave the old metric_arg class

Reply:
in addition to the metric type, there's also ADVI metric - "meanfield" and "fullrank".

Reply:
granted, 2 examples is not enough.  were there a 3rd, maybe.

Reply:
reverted to using classes "arg_metric" for HMC metric and "arg_variational_algo" for choice of ADVI algorithm.
not worth fighting the underlying design.